src/api.v2/model/Experiment.js

```diff
@@ -68,17 +69,33 @@ class Experiment extends BasicModel {
       .leftJoin(`${tableNames.EXPERIMENT_PARENT} as p`, 'e.id', 'p.experiment_id')
       .as('mainQuery');
 
-    const result = await collapseKeyIntoArray(
+    const experiments = await collapseKeyIntoArray(
       mainQuery,
       [...fields, 'parent_experiment_id', 'is_subsetted'],
       'key',
       'metadataKeys',
       this.sql,
     );
 
-    return result;
+    // Assuming the getMetadataByExperimentId function is async and returns the metadata
+    const cellLevel = new CellLevel();
+    const experimentIds = experiments.map((experiment) => experiment.id);
+    const cellLevelResults = await cellLevel.getMetadataByExperimentIds(experimentIds);
+
+    if (cellLevelResults.length) {
```

Comment:
This `if` shouldn't be necessary, if length === 0 then `forEach` doesn't do anything (there's nothing to iterate over).